src/t_stream.c

```diff
@@ -3608,6 +3988,106 @@ void xdelCommand(client *c) {
     if (ids != static_ids) zfree(ids);
 }
 
+/* Used by xdelexCommand() */
+typedef enum XDelexRes {
+    XDELEX_NO_ID = -1,           /* ID not found in the stream. */
+    XDELEX_DELETED = 0,          /* Message deleted. */
+    XDELEX_STILL_REFERENCED = 1, /* Message not deleted (still referenced). */
+} XDelexRes;
+
+/* XDELEX <key> [KEEPREF|DELREF|ACKED] [IDS <numids> <id ...>]
+ *
+ * Removes specified entries from the stream. Returns an array of status codes for
+ * each ID, indicating whether it was deleted, still referenced, or not found. */
+void xdelexCommand(client *c) {
+    kvobj *kv = lookupKeyWrite(c->db, c->argv[1]); 
+    if (checkType(c, kv, OBJ_STREAM)) return;
+
+    /* Parse command options */
+    streamAckDelArgs args;
+    if (!streamParseAckDelArgsOrReply(c, 2, &args)) return;
+
+    /* Non-existing keys and empty stream are the same thing. Reply null if the
+     * key does not exist.*/
+    if (!kv) {
+        addReplyArrayLen(c, args.numids);
+        for (int i = 0; i < args.numids; i++)
+            addReplyNull(c);
```

Comment:
can't we reply with XDELEX_NO_ID? or maybe some other integer XDELEX_NO_KEY? Because in other similar PRs, we didn't change the type in the array. See `httl` command, it does not reply with nulls.